src/core/ext/transport/chttp2/transport/writing.cc

```diff
@@ -446,6 +470,13 @@ class StreamWriteContext {
         "send_initial_metadata_finished");
   }
 
+  bool compressed_data_buffer_len() {
```

Comment:
would it be better to just initialize compressed_data_buffer so that we get a length of 0 to avoid the comparison?

Reply:
That's essentially what I'm trying to avoid here. I've moved these slice buffers to the end of the struct to avoid polluting the cache lines and I'd like to now touching the buffers unless necessary.